### PR TITLE
Use golangci-lint 1.46.2

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/kubewarden-controller/issues/217

Use latest for golangci-lint, now that there's a freshly released version. Fix small lint errors, disable exhaustruct as we already did with exhaustivestruct.

## Test

CI, and locally with `make lint`.

### CODE REVIEW

**Comment by jvanz** [comment]
Is the failing lint CI expected?

**Comment by raulcabello** [comment]
lint is failing in [CI](https://github.com/kubewarden/kubewarden-controller/runs/6471357321?check_suite_focus=true) as @jvanz mentioned.

**Comment by viccuad** [comment]
re-run the job, it works fine (I think it was a GHA cache issue. It works fine with 1.46.2 locally).

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -22,4 +22,4 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.45.2
+          version: latest
```
Comment (flavio) [inline .github/workflows/golangci-lint.yml:25]: I have mixed feelings about not pinning a specific version. I fear the CI could start to break out of the blue as soon as a new version of golangci-lint is released.

However, feel free to go ahead. Do **not** consider this comment as blocking the merge

**Review by flavio — APPROVED** [review]
LGTM, I left an open question.

Also, please rebase against `main`, it's not possible to merge the PR right now

**Review by jvanz — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]

**Comment by viccuad** [comment]
rebased onto main twice, because I'm silly, and went around merging other PRs meanwhile this CI finished..

**Pull request merged**